src/Microsoft.ML.TimeSeries/FftUtils.cs

```diff
@@ -0,0 +1,295 @@
+
+            /* Number of user threads that share the descriptor [1] */
+            NumberOfUserThreads = 26
+        }
+
+        private enum ConfigValue
+        {
+            /* CommitStatus */
+            Committed = 30,
+            Uncommitted = 31,
+
+            /* ForwardDomain */
+            Complex = 32,
+            Real = 33,
+            /* ConjugateEven = 34,   ## NOT IMPLEMENTED */
+
+            /* Precision */
+            Single = 35,
+            Double = 36,
+
+            /* ForwardSign */
+            /* Negative = 37,         ## NOT IMPLEMENTED */
+            /* Positive = 38,         ## NOT IMPLEMENTED */
+
+            /* ComplexStorage and ConjugateEvenStorage */
+            ComplexComplex = 39,
+            ComplexReal = 40,
+
+            /* RealStorage */
+            RealComplex = 41,
+            RealReal = 42,
+
+            /* Placement */
+            InPlace = 43,          /* Result overwrites input */
+            NotInPlace = 44,      /* Have another place for result */
+
+            /* InitializationEffort */
+            /* Low = 45,              ## NOT IMPLEMENTED */
+            /* Medium = 46,           ## NOT IMPLEMENTED */
+            /* High = 47,             ## NOT IMPLEMENTED */
+
+            /* Ordering */
+            Ordered = 48,
+            BackwardScrambled = 49,
+            /* ForwardScrambled = 50, ## NOT IMPLEMENTED */
+
+            /* Allow/avoid certain usages */
+            Allow = 51,            /* Allow transposition or workspace */
+            /* Avoid = 52,            ## NOT IMPLEMENTED */
+            None = 53,
+
+            /* PackedFormat (for storing congugate-even finite sequence
+               in real array) */
+            CcsFormat = 54,       /* Complex conjugate-symmetric */
+            PackFormat = 55,      /* Pack format for real DFT */
+            PermFormat = 56,      /* Perm format for real DFT */
+            CceFormat = 57        /* Complex conjugate-even */
+        }
+
+        private const string DllName = "MklImports";
+
+        // See: https://software.intel.com/en-us/node/521976#8CD904AB-244B-42E4-820A-CC2376E776B8
+        [DllImport(DllName, EntryPoint = "DftiCreateDescriptor")]
+        private static extern int CreateDescriptor(out IntPtr desc, ConfigValue precision, ConfigValue domain, int dimension, int length);
+
+        // See: https://software.intel.com/en-us/node/521977
+        [DllImport(DllName, EntryPoint = "DftiCommitDescriptor")]
+        private static extern int CommitDescriptor(IntPtr desc);
+
+        // See: https://software.intel.com/en-us/node/521978
+        [DllImport(DllName, EntryPoint = "DftiFreeDescriptor")]
+        private static extern int FreeDescriptor(ref IntPtr desc);
+
+        // See: https://software.intel.com/en-us/node/521981
+        [DllImport(DllName, EntryPoint = "DftiSetValue")]
+        private static extern int SetValue(IntPtr desc, ConfigParam configParam, ConfigValue configValue);
```

Comment:
I spent almost two days in trying to figure out what to do with arglist params, which no longer supported for coreclr.

[https://software.intel.com/en-us/mkl-developer-reference-c-dftisetvalue](https://software.intel.com/en-us/mkl-developer-reference-c-dftisetvalue) 
MKL_LONG DftiSetValue( DFTI_DESCRIPTOR_HANDLE, DFTI_CONFIG_PARAM , ... );
that ... used to be supported through __arglist.

At some point I gave up and wrote proxy c++ code.
So if your tests start failing on Linux/Mac only, and you don't know why, this is probably the case.

Reply:
thanks, @Ivanidzo4ka